### PR TITLE
update commitment contract

### PR DESCRIPTION
**Pull request by daiki44**
- [x] submit
- [x] getCurrentBlock
- [x] getRoot

### CODE REVIEW

File: packages/contract/src/CommitmentContract.ts [hunk 1]
```diff
@@ -1,34 +1,104 @@
...
   subscribeBlockSubmitted(
     handler: (blockNumber: BigNumber, root: Bytes) => void
   ) {
-    throw new Error('Not implemented')
+    this.eventWatcher.subscribe(EventType.BLOCK_SUBMITED, (log: EventLog) => {
+      const blockNumber = log.values[0].int
+      const root = log.values[1].bytes
+      handler(
+        BigNumber.fromString(blockNumber.toString()),
+        // remove 05
```
Comment (daiki44) [inline packages/contract/src/CommitmentContract.ts]: @syuhei176 Do you think when should I remove Tezos bytes prefix?
For now, I remove the prefix when subscribing to the event.

Comment (syuhei176) [reply packages/contract/src/CommitmentContract.ts]: I think Bytes from Wakkanai can accept '0x' prefix, so you need not to remove.

Comment (daiki44) [reply packages/contract/src/CommitmentContract.ts]: Sorry. Tezos bytes prefix = `05` not `0x`.

Comment (syuhei176) [reply packages/contract/src/CommitmentContract.ts]: eventItem is unpacked [here](https://github.com/cryptoeconomicslab/wakkanay-tezos/pull/19/files/261376cdefc91ee12f88ea92a007d57897b1039d#diff-f60b29ca1ffd2d53322ce5fe8af6a5b7R139)
So, I was thinking `root` wasn't bytes_packed. when the root byte packed?

Comment (daiki44) [reply packages/contract/src/CommitmentContract.ts]: > eventItem is unpacked here
I see! So I guess we should remove the prefix (05) when unpacking the data.

> So, I was thinking root wasn't bytes_packed. when the root byte packed?
The root byte will be packed in the contract!
https://github.com/cryptoeconomicslab/ovm.tz/blob/master/contracts/actions/commitment.ligo#L30

Comment (syuhei176) [reply packages/contract/src/CommitmentContract.ts]: Can you confirm actual data by unit test? I think root is unpacked bytes(not packed data). So it must not have prefix.

```
const root = log.values[1].bytes
```

File: packages/contract/__tests__/events/testInitialStorage.ts [hunk 2]
```diff
@@ -141,7 +141,7 @@ export const testEventStorage: MichelinePrim = {
                       prim: 'Pair',
                       args: [
                         { int: '0' },
-                        [{ bytes: '001' }, { bytes: '0100000004726f6f74' }]
+                        [{ bytes: '001' }, { bytes: '0a00000006010203040506' }]
```
Comment (syuhei176) [inline packages/contract/__tests__/events/testInitialStorage.ts]: I don't know it's correct or not, but storage is like below by current main branch's `npm run build:json`.

```
[ { "string": "BlockSubmitted" },
                          [ { "prim": "Pair",
                              "args":
                                [ { "int": "0" },
                                  [ { "bytes": "050000" },
                                    { "bytes": "050a00000006010203040500" } ] ] } ] ]
```

**Comment by syuhei176** [comment]
@daiki44 can we merge this PR?

**Comment by daiki44** [comment]
> @daiki44 can we merge this PR?

Sorry I have not tested yet.

**Pull request merged**